src/content/blog/advanced-add-on-contributor-training.md

```diff
@@ -0,0 +1,188 @@
+
+```yaml
+project_files:
+  - commands/web/eslint
+  - commands/web/expand-composer-json
+  - commands/web/nightwatch
+  - commands/web/phpcbf
+  - commands/web/phpcs
+  - commands/web/phpunit
+  - commands/web/poser
+  - commands/web/stylelint
+  - commands/web/symlink-project
+```
+
+You can add to the `global_files` section in the same way, but it usually makes more sense to add commands to the project, since the add-on's scope is project-level. [`ddev-platformsh`](https://github.com/ddev/ddev-platformsh) installs a helper command globally:
+
+```yaml
+global_files:
+  - commands/web/platform
+```
+
+## Adding a `Dockerfile.<add-on-name>`
+
+If your `install.yaml` needs to change basic `ddev-webserver` characteristics, it can add a `.ddev/web-build/Dockerfile.<add-on-name>`.
+
+[`envsa/ddev-pnpm`](https://github.com/envsa/ddev-pnpm/blob/main/install.yaml#L11-L13) does exactly this:
+
+```yaml
+project_files:
+  - commands/web/pnpm
+  - web-build/Dockerfile.pnpm
+```
+
+## Altering the behavior of `ddev-webserver` with additional `config.*.yaml`
+
+An add-on can deliver both project and global configuration, and can add to or replace almost anything. For example, the [`justafish/ddev-drupal-core-dev`](https://github.com/justafish/ddev-drupal-core-dev) add-on installs a `config.ddev-drupal-core-dev.yaml` that does a few things, including:
+
+```yaml
+webimage_extra_packages: ["chromium-driver"]
+```
+
+## Altering the behavior of `ddev-webserver` with `docker-compose.*.yaml`
+
+Although the most common use of adding a `docker-compose.*.yaml` is to add a new service (below) it is often used to change the behavior of the web service. 
+
+For example, [`ddev-proxysupport`](https://github.com/ddev/ddev-proxy-support/blob/7c8a91fb020bf2df62418730352d3db5a1ca76e3/docker-compose.proxy-support.yaml#L3-L10) sets arguments for the `build`stage inside the web container.
+
+## Creating an additional service using a `docker-compose.*.yaml`
+
+One of the most common add-on uses is to create a new service, like `mongo` or `elasticsearch` or `solr`. You can look at many of the official add-ons to see how this is done.
+
+Examples:
+
+* [ddev-solr](https://github.com/ddev/ddev-solr/docker-compose.solr.yaml)
+* [ddev-memcached](https://github.com/ddev/ddev-memcached/blob/main/docker-compose.memcached.yaml)
+
+See the [general docs on extra services](https://ddev.readthedocs.io/en/stable/users/extend/custom-compose-files/#third-party-services-may-need-to-trust-ddev-webserver)
+
+## Interacting with users during `install.yaml` installs
+
+Although unusual, it is sometimes useful to interact with the user during the `ddev get` process. For example, [`ddev-platformsh`](https://github.com/ddev/ddev-platformsh) checks to make sure that the `PLATFORMSH_CLI_TOKEN` has been properly configured, and, if not, requests it and configures it:
+
+```yaml
+    #ddev-nodisplay
```

Comment:
````suggestion
```yaml
pre_install_actions:
  # Get PLATFORMSH_CLI_TOKEN from user if we don't have it yet
  - |
    #ddev-nodisplay
````